nominal/core.py

```diff
@@ -311,6 +339,132 @@ def _from_conjure(cls, client: NominalClient, dataset: scout_catalog.EnrichedDat
         )
 
 
+@dataclass(frozen=True)
+class LogSet:
+    rid: str
+    created_by: str
+    name: str
+    created_at: datetime
+    updated_at: datetime
+    log_count: int
+    timestamp_type: TimestampType
+    origin_metadata: Mapping[str, str]
+    description: str | None
+    _client: NominalClient = field(repr=False)
+
+    def add_logs_and_finalize(self, logs: list[Log]) -> Self:
+        """
+        Attaches a list of logs to this log set and finalizes it. A logset is not considered readable
+        until it has been finalized. Once finalized, the logset is immutable.
+        """
+        request = datasource_logset_api.AttachLogsAndFinalizeRequest(
+            logs=[log._to_conjure() for log in logs],
+        )
+        response = self._client._logset_client.attach_logs_and_finalize(
+            auth_header=self._client._auth_header, log_set_rid=self.rid, request=request
+        )
+        log_set_metadata = self.__class__._from_conjure(self._client, response)
+        update_dataclass(self, log_set_metadata, fields=self.__dataclass_fields__)
+        return self
+
+    def get_logs_paginated(
+        self,
+        page_size: int | None = None,
+    ) -> PaginatedLogs:
+        """
+        Get logs within this log set. Paginated, maximum page size is 10000, defauls to maximum.
+        """
+        response = self._client._logset_client.search_logs(
+            self._client._auth_header,
+            log_set_rid=self.rid,
+            request=datasource_logset_api.SearchLogsRequest(page_size=page_size),
+        )
+        return PaginatedLogs(
+            logs=[Log._from_conjure(log) for log in response.logs],
+            page_size=page_size,
+            next_page_token=response.next_page_token,
+            client=self._client,
+        )
+
+    @classmethod
+    def _from_conjure(cls, client: NominalClient, log_set_metadata: datasource_logset_api.LogSetMetadata) -> Self:
+        timestamp_type = None
+        if log_set_metadata.timestamp_type == datasource.TimestampType.ABSOLUTE:
+            timestamp_type_enum = "absolute"
+        elif timestamp_type_enum == datasource.TimestampType.RELATIVE:
+            timestamp_type_enum = "relative"
+        else:
+            raise ValueError(f"unsupported timestamp type: {timestamp_type}")
+
+        return cls(
+            rid=log_set_metadata.rid,
+            created_by=log_set_metadata.created_by,
+            name=log_set_metadata.name,
+            created_at=parser.parse(log_set_metadata.created_at),
+            updated_at=parser.parse(log_set_metadata.updated_at),
+            log_count=log_set_metadata.log_count,
+            timestamp_type=timestamp_type,
+            origin_metadata=MappingProxyType(log_set_metadata.origin_metadata),
+            description=log_set_metadata.description,
+            _client=client,
+        )
+
+
+@dataclass(frozen=True)
+class Log:
+    time: datetime | IntegralNanosecondsUTC
```

Comment:
lets standardize on "timestamp" as the name of every time field for the client lib

Reply:
renamed to timestamp